google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -468,23 +457,18 @@ void extendDeadlines() {
     List<PendingModifyAckDeadline> modacks = new ArrayList<>();
     PendingModifyAckDeadline modack = new PendingModifyAckDeadline(extendSeconds);
     Instant now = now();
-    Instant extendTo = now.plusSeconds(extendSeconds);
 
     for (Map.Entry<String, AckHandler> entry : pendingMessages.entrySet()) {
       String ackId = entry.getKey();
       Instant totalExpiration = entry.getValue().totalExpiration;
-      if (totalExpiration.isAfter(extendTo)) {
+      if (totalExpiration.isAfter(now)) {
```

Comment:
What was the reason for this change? This is changing the semantics of extending ack deadlines for messages that are near the total expiration. With this change, messages could be extended beyond their total expiration.

Reply:
This is part of #4559, and only exists in this pr due to the diffbase.  I believe the comment there explains my reasoning behind this change- let me know if you disagree and think I should revert that there.